### ISSUE TITLE
Bitcoinj/bitcoinj is being compiled with Java 8, breaks Android builds.

### ISSUE THREAD

**Issue by acrespo** [jitpack/jitpack.io#2159]
I'm trying build my android app with segwit branch as a dependency (there´s no release yet) but [it is being compiled with Java 8](https://jitpack.io/com/github/bitcoinj/bitcoinj/segwit-v0.12-gf6a7d22-929/build.log). Unfortunately, this breaks since android doesn't fully handle Java 8 yet. Perhaps there should be a way for users of the dependencies to choose which jdk to use. I know, it should be a very "use it at your own risk" feature but otherwise we are forced to ask repo owners to add their "jitpack.yml" file to enforce a particular jdk, or fork and do it ourselves (for each dependency).

**Comment by jitpack-io** [comment]
Hi @acrespo 

It seems BitcoinJ requires Java 8 since it's explicitly set in the pom: https://github.com/bitcoinj/bitcoinj/blob/master/pom.xml#L89
They might be using Java 8 features.

**Comment by acrespo** [comment]
Hi there! 

Thanks for the response! Yeah I saw that, but for the module I'm interested in (core) they are actually using Java 7. [Java 7](https://github.com/bitcoinj/bitcoinj/blob/master/core/pom.xml#L118).
I wonder if we could make Jitpack build different modules with different jdk versions and/or just compiled the modules we are interested in (I tried setting just the core module as a dependency but Jitpack tries to compile the whole project).
